src/main/java/by/custom/utilcalculator/domain/UserProgressStorage.java

```diff
@@ -1,27 +1,52 @@
 package by.custom.utilcalculator.domain;
 
+import org.telegram.telegrambots.meta.api.objects.Message;
+
+import java.io.*;
 import java.util.HashMap;
 
 public class UserProgressStorage {
-    private final HashMap<String, UserProgress> users;
+
+    private static class UserProgressStorageHolder {
+        private static final UserProgressStorage USER_PROGRESS_STORAGE = new UserProgressStorage();
+    }
+
+    private HashMap<String, UserProgress> users;
```

Comment:
Please, use abstraction instead implementation
```suggestion
    private Map<String, UserProgress> users;
```

Reply:
Sorry, bad habit. Fixed.